### ISSUE TITLE
Query on productionizing a TagUI project to run on Linux server - some thoughts

### ISSUE THREAD

**Issue by kensoh** [kelaberetiv/TagUI#989]
*source - LinkedIn message*

---

I've been looking at TagUi and RPA for Python.  If I were to use it on a linux machine in unattended mode, does that mean all UI interactions with browsers shall be headless? Should I be thinking of the same pattern where UiPath development is done via Studio using my PC/laptop, then having it packaged/deployed to a bot running on a VM? Appreciate your response. Thanks!

In other words, how do I productionize a TagUI project I built locally in a linux laptop as an unattended bot process on a linux server?

**Comment by kensoh** [comment]
On GitHub or the Telegram group chat would be better - https://t.me/rpa_chat  So other users in the community can learn and also contribute their experience. I'll probably reply here and create a GitHub issue with some of these sharings, then share to the chat group.

For Linux, you can do it in visible mode or headless mode. But for visible mode you really need a desktop Linux or if you Linux only has the command-line interface, will need to setup Xvbf for visible mode. In the visible mode, you can also use visual automation with image identifiers and OCR. But for Linux need to install the OpenCV and Tesseract separately.

Can you tell me more about your usage scenario?

For eg, the expected number of users, whether the Linux server is expected to be up all the time. The expected response time back etc. All will determine how you'll set it up. Also, how you will want to trigger / return back results also determines how you set up the Linux servers.

**Comment by kensoh** [comment]
Lemme know more and I'll share more what could be good ways to get to that outcome you want!
